pkg/multicluster/virtual_cluster.go

```diff
@@ -168,6 +171,7 @@ func FindVirtualClustersByLabels(ctx context.Context, c client.Client, labels ma
 	for _, secret := range secrets.Items {
 		vc, err := NewVirtualClusterFromSecret(secret.DeepCopy())
 		if err == nil {
+			vc.Metrics = metricsMap[vc.Name]
```

Comment:
Maybe this part should be moved into `NewVirtualClusterFromSecret`?

Reply:
Yes, this is a duplicate assignment that I will remove later